### ISSUE TITLE
Migrate kubernetes images to python 3.7+

### ISSUE THREAD

**Issue by mvdbeek** [galaxyproject/galaxy#11828]
https://github.com/galaxyproject/galaxy/pull/11826 reminded me that we need to do this for release 21.09, where we're going to drop Python 3.6 support. But there's no reason we couldn't already add a newer Python version now.

**Comment by rhpvorderman** [comment]
Might I suggest moving to debian:buster-slim as a base image? Debian buster has python 3.7 as default. Also debian is the de facto default base image for a lot of production images (nginx and postgres come to mind).

**Comment by rhpvorderman** [comment]
Alternatively, we could switch to using the python container. That is based on debian, but comes with python preinstalled. The python container has all versions available in a slim variant: 3.7-slim, 3.8-slim etc.

**Comment by rhpvorderman** [comment]
@mvdbeek I have made a [galaxy-compose](https://github.com/lumc/galaxy-compose) image based on the k8s image. I used the python:3.7-slim container as a base.

I can port that work over to the galaxy k8s image now (working on it). Using the python image makes it really easy to change the version of python at a later date. The base image for python is debian, so not much changes (almost none) from ubuntu as a base.
